### DISCUSSION TITLE
Dynamic polygone zone and object detection

### DISCUSSION THREAD

**Discussion by aqibsaeed** [roboflow/supervision#821] [Q&A]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

Hi, thank you for all your efforts! I want to `supervision` for object detection (OD), but my use case requires OD within in a dynamic polygone zone (e.g., on road objects where a car keep moving forward so zone being updated continuously). Could you please guide me to relevant functions that I can use to implement this? Thanks!

**Comment by SkalskiP** [comment]
Hi @aqibsaeed 👋🏻 Let me convert this question into a discussion and place it Q&A section.

**Comment by SkalskiP** [comment]
Is the road your zone?

**Comment by aqibsaeed** [comment]
Yes, the road could be a zone.

**Reply by SkalskiP** [reply]
Then, you need to run a multi-stage pipeline. Detect the road first and then use it as a zone.

**Reply by aqibsaeed** [reply]
Is there any example for multi-stage pipeline?

**Reply by SkalskiP** [reply]
Nope. We did not show any examples of multi-stage pipeline yet.

**Reply by aqibsaeed** [reply]
Ok, to implement it basically one has to instantiate two model and tracker objects, and have to call object detection model under callback of road detection, right?

**Reply by SkalskiP** [reply]
Yup, that's pretty much what you'd need to do.
